### DISCUSSION TITLE
Falha no carregamento da biblioteca

### DISCUSSION THREAD

**Discussion by jgmsantos** [AlertaDengue/PySUS#211] [Q&A] [answered]
Olá, pessoal. Tudo bem?

Estou tentando ler a base do SINAN de acordo com a imagem abaixo, mas sem sucesso, fica só retornando timeout.

![fig02](https://github.com/user-attachments/assets/9c024680-3a5c-48ba-a7b7-5333b565a01f)

Estou usando:

!pip install pysus

from google.colab import drive
drive.mount('/content/drive', force_remount=True)
from pysus.online_data import SINAN

Eu já usava essa estrutura e funcionava sem problemas.

Alguém poderia dar uma ajuda?

**Comment by felipeadeildo** [comment, marked as answer]
Opa, boa tarde.

Então, basicamente o módulo "online_data" funciona em cima do FTP do datasus, então quando você importa a classe SINAN do módulo "online_data", ele define um objeto chamado "Directory" que ao ser instanciado cria uma conexão FTP com o endrećo "ftp.datasus.gov.br" ([ver aqui](https://github.com/AlertaDengue/PySUS/blob/main/pysus/ftp/__init__.py#L239))

Porém, como trata0se de dados do Governo Brasileiro e dados públicos para a populaćão, não faz sentido ele estar aberto para IP que não seja Brasileiro.

Quando vc faz a importaćão utilizando IP estrangeiro (como é o caso do ambiente do Colaboratory), o servidor nega a conexão resultando num timeout (sem resposta do servidor a IP`s de fora do Brasil).

Em resumo, é isso.

Como resolver? Roda isso localmente e seja feliz :D

**Reply by jgmsantos** [reply]
Olá, Felipe. Tudo bem? 

Faz todo o sentido sua explicação, desde ontem estava tentando baixar os dados e agora consegui fazer isso. 

Obrigado pela resposta.

Abraços.

**Comment by fccoelho** [comment]
Obrigado pelo input @felipeadeildo, é assim mesmo que o PySUS funciona. Mas esta restrição a IPs não brasileiros deve ser uma coisa nova, pois quando criei os notebooks de exemplo que estão públicos no Colab, este acesso funcionava sem problema.

Para os nossos usuários que usam Windows, sugiro familiarizarem-se com o WSL, e instalarem o PySUS neste ambiente.

**Reply by jgmsantos** [reply]
Eu estranhei por isso mesmo porque antes eu baixava os dados sem problemas e agora deu bo. De qualquer forma, obrigado por disponibilizar essas informações e pela ajuda.
